### DISCUSSION TITLE
Integrate edamame

### DISCUSSION THREAD

**Discussion by jeroenvandijk** [babashka/babashka#1129] [Ideas]
**Is your feature request related to a problem? Please describe.**

Babashka is missing some clojure.core vars in order to support Edamame dynamically see  https://github.com/borkdude/sci/issues/382 https://github.com/borkdude/sci/issues/383

Edamame is useful to parse clojure forms in clojure files. I was trying this because I want to find duplicated functions in my code via a Babashka script. And I'll probably think of more applications when this is succesful

**Describe the solution you'd like**

Be able to load the library edamame without errors:

```
export BABASHKA_CLASSPATH=$(clojure -Spath -Sdeps '{:deps {  borkdude/edamame {:mvn/version "0.0.11-alpha.13"} }  }')
```

```clojure
(require '[edamame.core :refer [parse-string]]))
```

**Describe alternatives you've considered**

- Using another library https://github.com/babashka/pod-babashka-parcera
- Writing the script in Clojure
- Not writing such a script

**Additional context**

The two issues mentioned in the problem statement above

**Comment by borkdude** [comment]
@jeroenvandijk I'd like to not expose edamame just yet, since babashka is pretty conservative in what is built-in. 

But I have good news. You can parse clojure using `clojure.core/read` in babashka now, which uses edamame behind the scenes.

E.g this shows there are 689 forms in clojure.core:

```
$ bb -e '(let [rdr (io/reader "/Users/borkdude/git/clojure/src/clj/clojure/core.clj") rdr (java.io.PushbackReader. rdr)] (count (take-while #(not= ::eof %) (repeatedly #(read {:eof ::eof} rdr)))))'
698
```

**Comment by borkdude** [comment]
I've also used this approach in the tools.namespace fork here: https://github.com/babashka/tools.namespace

**Reply by borkdude** [reply]
Forgot to mention: babashka now also comes with rewrite-clj which may serve a similar purpose.

**Comment by borkdude** [comment]
@jeroenvandijk Edamame is now (finally!) available in babashka on master.

**Reply by jeroenvandijk** [reply]
@borkdude Thank you!
